src/twisted/logger/test/_assertionhelpers.py

```diff
@@ -0,0 +1,244 @@
+    """
+    Assertion helper for testing a single log event that has been
+    emitted using L{twisted.logger._loggerFor}, in which case
+    the L{Logger} that the event was emitted to may not be available
+    for testing, hence no argument for log_logger. This helper should
+    preferably be used in conjunction with the context manager
+    L{capturingLogEventsFromGlobalLogPublisher} for optimum test
+    coverage for events that can only be tested by passing an observer
+    to a L{globalLogPublisher}.
+
+    @param testCase: The L{TestCase} object that wants to test some
+        log events using L{assertLogEvent}.
+    @type testCase: L{TestCase}
+
+    @param event: The emitted event, which is a L{dict}.
+    @type event: L{dict}
+
+    @param log_level: A L{LogLevel}.
+    @type log_level: L{LogLevel}
+
+    @param log_namespace: The namespace associated with the L{Logger}.
+    @type log_namespace: L{str} (native string)
+
+    @param log_source: The source object that emitted the event. This
+        will be L{None} if the L{Logger} is not accessed as an
+        attribute of an instance or class.
+    @type log_source: L{object} or L{None}
+
+    @param log_format: The format string provided for use by observers
+        that wish to render the event as text. The format string uses
+        new-style PEP 3101 formatting and is rendered using the log
+        event (which is a L{dict}). This may be L{None}, if no format
+        string was provided. The format string is optional, except when
+        a failure is being logged using L{Logger.failure}, in which
+        case a format string describing the failure must be provided.
+    @type log_format: L{str} or L{None}
+
+    @param kwargs: Additional key/value pairs that we expect to see
+        in the event being tested. This L{dict} may contain key/value
+        pairs needed to effectively render the format string.
+    @type kwargs: L{dict}
+    """
+    expectedEvent = {
+        'log_level': log_level, 'log_source': log_source,
+        'log_format': log_format, 'log_namespace': log_namespace
+    }
+    expectedEvent.update(kwargs)
+    expectedEvent.update(log_time=event['log_time'])
+    if 'log_failure' in event:
+        expectedEvent.update(log_failure=event['log_failure'])
+
+    # Since we can't test log_logger, treat it like log_time
+    expectedEvent.update(log_logger=event['log_logger'])
+
+    # Compatibility with the old logging system, for more on this,
+    # see twisted.logger._legacy.LegacyLogObserverWrapper and
+    # twisted.logger._legacy.ensureCompatWithLegacyObservers
+    expectedEvent = _ensureCompatWithLegacyObservers(expectedEvent)
+    if event['log_format'] is not None:
+        expectedEvent.update(log_legacy=event['log_legacy'])
+
+    testCase.maxDiff = None
+
+    testCase.assertEqual(event, expectedEvent)
+
+
+
+def assertLogEvent(testCase, event, log_logger, log_level, log_namespace,
```

Comment:
This is a very long signature!  But I'm not sure if it can be any shorter.

If `capturingLogEventsFrom'`s API changes according to my suggestion above, you can at least remove the `log_logger` argument.